src/main/java/de/plushnikov/intellij/plugin/hack/extension/EMHandler.java

```diff
@@ -0,0 +1,267 @@
+        @Nullable PsiClass context = PsiTreeUtil.getParentOfType(place, PsiClass.class);
+        while (context != null) {
+          final @Nullable PsiAnnotation annotation = context.getAnnotation(LombokClassNames.EXTENSION_METHOD);
+          if (annotation != null) {
+            final Set<PsiClass> providers = PsiAnnotationUtil.getAnnotationValues(annotation, "value", PsiType.class).stream()
+              .filter(PsiClassType.class::isInstance)
+              .map(PsiClassType.class::cast)
+              .map(PsiClassType::resolve)
+              .filter(Objects::nonNull)
+              .collect(Collectors.toSet());
+            if (!providers.isEmpty()) {
+              final Collection<PsiMethod> collect = supers(psiClass).map(PsiClassUtil::collectClassMethodsIntern).flatMap(Collection::stream).collect(Collectors.toSet());
+              final Map<PsiClass, Collection<PsiMethod>> providerRecord = new HashMap<>();
+              PsiType type = null;
+              if (place instanceof PsiMethodCallExpression) {
+                final @Nullable PsiExpression expression = ((PsiMethodCallExpression) place).getMethodExpression().getQualifierExpression();
+                if (expression != null)
+                  type = expression.getType();
+              } else if (place instanceof PsiReferenceExpression && ((PsiReferenceExpression) place).getQualifier() != null)
+                type = ((PsiReferenceExpression) place).getQualifierExpression().getType();
+              if (type == null)
+                type = PsiTypesUtil.getClassType(psiClass);
+              final PsiType finalType = type;
+              final boolean match = (name == null ?
+                injectAugment(providers, finalType, psiClass) :
+                supers(psiClass).flatMap(node -> injectAugment(providers, psiClass == node ? finalType : new PsiImmediateClassType(node, PsiSubstitutor.EMPTY), node))
+                  .filter(method -> name.equals(method.getName())))
+                .filter(method -> checkMethod(collect, providerRecord, method))
+                .allMatch(method -> processor.execute(method, state));
+              if (!match)
+                return false;
+            }
+          }
+          context = PsiTreeUtil.getParentOfType(context, PsiClass.class);
+        }
+      }
+    }
+    return true;
+  }
+
+  private static Stream<PsiMethod> injectAugment(final Set<PsiClass> providers, final PsiType type, final PsiClass node) {
+    return providers.stream().map(EMHandler::providerData)
+      .map(Map::entrySet)
+      .flatMap(Set::stream)
+      .filter(e -> checkType(e.getKey(), type, node))
+      .map(Map.Entry::getValue)
+      .flatMap(List::stream)
+      .map(function -> {
+        try {
+          return function.apply(node, type);
+        } catch (Throwable throwable) {
+          if (throwable instanceof ProcessCanceledException)
+            throw throwable;
+          throwable.printStackTrace();
```

Comment:
I think to use "printStackTrace" is not a very optimal  solution

Reply:
Oh, yes, this try block I should remove, actually no exceptions occur here, it's a legacy I used to test cache consistency before.